tau = MC/hA. a small temperature probe

a) find time constant in air and oil;

b) in stirred oil find -3dB cutoff angular frequency and frequency in Hz. state the useful frequency range

c) the probe initially reads 20 in air, and its suddenly placed at oil(80) how long does it takes to read 70

d) immediately after reaching 70, then suddenly move to air at 25. what is the indicate temperature 60s later


In [16]:
import numpy as np

# Given parameters
M = 0.015       # kg
C = 390         # J/(kg·K), specific heat
A = 2e-4        # m^2, surface area

# Heat transfer coefficients (W/(m^2·K))
h_air = 35      # natural convection in still air
h_oil = 350     # forced convection in stirred oil

# Part (a): Time constants
tau_air = M * C / (h_air * A)
tau_oil = M * C / (h_oil * A)

print(f"=== Part (a) ===")
print(f"τ_air = {tau_air:.2f} s  ({tau_air/60:.2f} min)")
print(f"τ_oil = {tau_oil:.2f} s")
print(f"\nThe probe responds ~{tau_air/tau_oil:.1f}x faster in stirred oil than in still air.")

=== Part (a) ===
τ_air = 835.71 s  (13.93 min)
τ_oil = 83.57 s

The probe responds ~10.0x faster in stirred oil than in still air.


## Part (b): -3dB Cutoff Frequency in Stirred Oil

For a first-order system, the transfer function is $H(s) = \frac{1}{\tau s + 1}$.

At $\omega = \omega_c$, $|H(j\omega_c)| = \frac{1}{\sqrt{2}}$ (-3dB point).

$$\omega_c = \frac{1}{\tau}, \qquad f_c = \frac{\omega_c}{2\pi} = \frac{1}{2\pi\tau}$$

The **useful frequency range** is where the amplitude attenuation is negligible: typically $f \ll f_c$ (one decade or more below cutoff).

In [17]:
# Part (b): -3dB cutoff frequencies in stirred oil
omega_c = 1 / tau_oil        # angular cutoff frequency (rad/s)
f_c = omega_c / (2 * np.pi)  # cutoff frequency in Hz

print(f"=== Part (b) ===")
print(f"τ_oil = {tau_oil:.2f} s")
print(f"ω_c (angular cutoff, -3dB) = {omega_c:.6f} rad/s")
print(f"f_c (cutoff frequency, -3dB) = {f_c:.6f} Hz")
print(f"\nUseful frequency range:")
print(f"  f << f_c  →  f << {f_c:.6f} Hz")
print(f"  Practically, f ≤ {f_c/10:.6f} Hz (one decade below cutoff)")
print(f"  i.e., near-DC up to ~{f_c/10*1000:.3f} mHz")
print(f"\nAttenuation at various fractions of f_c:")
for ratio in [0.01, 0.05, 0.1, 0.5, 1.0]:
    mag = 1 / np.sqrt(1 + ratio**2)
    dB = 20 * np.log10(mag)
    print(f"  f = {ratio*f_c:.6f} Hz ({ratio*100:4.0f}% of f_c): |H| = {mag:.4f} ({dB:+.2f} dB)")

=== Part (b) ===
τ_oil = 83.57 s
ω_c (angular cutoff, -3dB) = 0.011966 rad/s
f_c (cutoff frequency, -3dB) = 0.001904 Hz

Useful frequency range:
  f << f_c  →  f << 0.001904 Hz
  Practically, f ≤ 0.000190 Hz (one decade below cutoff)
  i.e., near-DC up to ~0.190 mHz

Attenuation at various fractions of f_c:
  f = 0.000019 Hz (   1% of f_c): |H| = 1.0000 (-0.00 dB)
  f = 0.000095 Hz (   5% of f_c): |H| = 0.9988 (-0.01 dB)
  f = 0.000190 Hz (  10% of f_c): |H| = 0.9950 (-0.04 dB)
  f = 0.000952 Hz (  50% of f_c): |H| = 0.8944 (-0.97 dB)
  f = 0.001904 Hz ( 100% of f_c): |H| = 0.7071 (-3.01 dB)


## Part (c): Step Response — Air (20°C) → Oil (80°C)

First-order step response:
$$T(t) = T_\infty + (T_0 - T_\infty) e^{-t/\tau}$$

We want $T(t) = 70^\circ\text{C}$ in oil ($\tau_\text{oil} = 83.57$ s):

$$70 = 80 + (20 - 80) e^{-t/\tau_\text{oil}}$$
$$e^{-t/\tau_\text{oil}} = \frac{80-70}{80-20} = \frac{1}{6}$$
$$t = \tau_\text{oil} \cdot \ln(6)$$

In [18]:
# Part (c): Time to reach 70°C when plunged from 20°C air into 80°C oil
T0_c = 20      # initial temp in air (°C)
T_inf_c = 80   # oil temperature (°C)
T_target = 70  # target reading (°C)

t_to_70 = tau_oil * np.log((T_inf_c - T0_c) / (T_inf_c - T_target))

print(f"=== Part (c) ===")
print(f"T(t) = T∞ + (T0 - T∞)·exp(-t/τ)")
print(f"70 = 80 + (20-80)·exp(-t/{tau_oil:.2f})")
print(f"exp(-t/τ) = (80-70)/(80-20) = 1/6")
print(f"t = τ·ln(6) = {tau_oil:.2f} · ln(6)")
print(f"\nTime to reach {T_target}°C: t = {t_to_70:.2f} s ({t_to_70/60:.2f} min)")

# Verify
T_check = T_inf_c + (T0_c - T_inf_c) * np.exp(-t_to_70 / tau_oil)
print(f"Verify: T({t_to_70:.2f}s) = {T_check:.1f}°C ✓")

# Show the full response curve at key points
print(f"\nTemperature at key times:")
for t_pt in [10, 30, 60, t_to_70, 150, 200, 300]:
    T_pt = T_inf_c + (T0_c - T_inf_c) * np.exp(-t_pt / tau_oil)
    print(f"  t = {t_pt:7.1f}s → T = {T_pt:6.2f}°C")

=== Part (c) ===
T(t) = T∞ + (T0 - T∞)·exp(-t/τ)
70 = 80 + (20-80)·exp(-t/83.57)
exp(-t/τ) = (80-70)/(80-20) = 1/6
t = τ·ln(6) = 83.57 · ln(6)

Time to reach 70°C: t = 149.74 s (2.50 min)
Verify: T(149.74s) = 70.0°C ✓

Temperature at key times:
  t =    10.0s → T =  26.77°C
  t =    30.0s → T =  38.10°C
  t =    60.0s → T =  50.73°C
  t =   149.7s → T =  70.00°C
  t =   150.0s → T =  70.03°C
  t =   200.0s → T =  74.52°C
  t =   300.0s → T =  78.34°C


## Part (d): Step Response — Oil (70°C) → Air (25°C), 60s later

After reaching 70°C in oil, the probe is suddenly moved to 25°C air.

In air, $\tau_\text{air} = 1170$ s. After $t = 60$ s in air:

$$T(60) = T_\infty + (T_0 - T_\infty) e^{-t/\tau_\text{air}}$$
$$T(60) = 25 + (70 - 25) e^{-60/1170}$$

Since $\tau_\text{air} \gg 60$s, the probe cools very slowly in still air.

In [19]:
# Part (d): Temperature 60s after moving from 70°C oil to 25°C air
T0_d = 70      # initial temp when moved to air (°C)
T_inf_d = 25   # air temperature (°C)
t_d = 60       # time elapsed in air (s)

T_after_60s = T_inf_d + (T0_d - T_inf_d) * np.exp(-t_d / tau_air)

print(f"=== Part (d) ===")
print(f"Starting temp: {T0_d}°C")
print(f"Air temp: {T_inf_d}°C")
print(f"τ_air = {tau_air:.1f} s")
print(f"\nT(60) = {T_inf_d} + ({T0_d}-{T_inf_d})·exp(-{t_d}/{tau_air:.0f})")
print(f"T(60) = {T_inf_d} + {T0_d-T_inf_d}·exp({-t_d/tau_air:.4f})")
print(f"T(60) = {T_inf_d} + {T0_d-T_inf_d}·{np.exp(-t_d/tau_air):.4f}")
print(f"\nAfter {t_d}s in {T_inf_d}°C air: T = {T_after_60s:.2f}°C")
print(f"Temperature drop = {T0_d - T_after_60s:.2f}°C (only ~{T0_d - T_after_60s:.1f}°C in 1 minute)")

# Show cooling trend
print(f"\nCooling trend in air:")
for t_pt in [30, 60, 120, 300, 600, 1170]:
    T_pt = T_inf_d + (T0_d - T_inf_d) * np.exp(-t_pt / tau_air)
    print(f"  t = {t_pt:6.0f}s → T = {T_pt:6.2f}°C", end="")
    if t_pt == tau_air:
        print(f"  (τ_air, ~63% to equilibrium)", end="")
    print()

# Time constant check: after 1τ, should be ~63% of way from 70 to 25
T_1tau = T_inf_d + (T0_d - T_inf_d) * np.exp(-1)
print(f"\nAfter 1τ ({tau_air:.0f}s): T = {T_1tau:.2f}°C")
print(f"  (63.2% of 45°C range: 25 + 45×0.368 = {T_1tau:.2f}°C)")

=== Part (d) ===
Starting temp: 70°C
Air temp: 25°C
τ_air = 835.7 s

T(60) = 25 + (70-25)·exp(-60/836)
T(60) = 25 + 45·exp(-0.0718)
T(60) = 25 + 45·0.9307

After 60s in 25°C air: T = 66.88°C
Temperature drop = 3.12°C (only ~3.1°C in 1 minute)

Cooling trend in air:
  t =     30s → T =  68.41°C
  t =     60s → T =  66.88°C
  t =    120s → T =  63.98°C
  t =    300s → T =  56.43°C
  t =    600s → T =  46.95°C
  t =   1170s → T =  36.10°C

After 1τ (836s): T = 41.55°C
  (63.2% of 45°C range: 25 + 45×0.368 = 41.55°C)
